src/main/java/org/apache/maven/model/jdom/util/JDomUtils.java

```diff
@@ -114,7 +129,7 @@ private static int calcNewElementIndex(String name, Element root) {
 
     List<String> elementOrder = JDomCfg.getInstance().getElementOrder(root.getName());
     if (elementOrder == null) {
-      addIndex = max(0, root.getContentSize() - 1);
+      addIndex = max(0, getLastElementIndex(root) + 1);
```

Comment:
What if there are comments? Comments are no Elements. 
If I understand this code correctly, they won't get covered, wouldn't they?!

Excerpt of my.xml
```
<profiles>
  <!-- This profile is for production only -->
  <profile>
     <id>prod</id>
     ...
  </profile>
</profiles>
```

Adding a new profile at index 0 results in

```
<profiles>
  <!-- This profile is for production only -->
 <profile>
     <id>dev</id>
     ...
  </profile>
  <profile>
     <id>prod</id>
     ...
  </profile>
</profiles>
```

But maybe I'm wrong. Do you have a test for this case?

Reply:
These comments should not be removed unless You remove the entire profiles element. You may add integration tests covering that. There are already some integration tests verifying that comments aren't removed.